assets/app.js

```diff
@@ -1,13 +1,138 @@
- * (and its CSS file) in your base layout (base.html.twig).
- */
-
-// any CSS you import will output into a single css file (app.css in this case)
-import './styles/app.scss';
-
-// start the Stimulus application
-import './bootstrap';
-import 'flowbite';
+import "./styles/app.scss";
+import "./bootstrap";
+import "flowbite";
+
+
+function selectAll(selector) {
+    return Array.from(document.querySelectorAll(selector));
+}
+
+
+document.addEventListener("DOMContentLoaded", function () {
+    // Get All options
+    let options = selectAll(".option");
+    let categoryButtons = selectAll(".category-button");
+    let bulletsContainer = document.querySelector(".bullet-navigation");
+
+
+    const handleClick = (activeIndex) => {
+        const category = options[activeIndex].getAttribute("data-category");
+        const categoryOptions = options.filter(
+            (option) => option.getAttribute("data-category") === category
+        );
+        const startIndex = categoryOptions.indexOf(options[activeIndex]);
+
+
+        options.forEach((option) => {
+            option.style.display = "none";
+        });
+
+
+        //Display the option and the next 5 pictures
+        let imagesToDisplay = categoryOptions.slice(startIndex, startIndex + 6);
+
+
+        //If less than 6 get from the start
+        while (imagesToDisplay.length < 6) {
+            imagesToDisplay = imagesToDisplay.concat(
+                categoryOptions.slice(0, 6 - imagesToDisplay.length)
+            );
+        }
+
+
+        // Affiche les images à afficher
+        imagesToDisplay.forEach((option) => {
+            option.style.display = "flex";
+        });
+
+
+        // Met en surbrillance l'option sélectionnée
+        options.forEach((option, index) => {
+            if (index === activeIndex) {
+                option.classList.add("active");
+            } else {
+                option.classList.remove("active");
+            }
+        });
+
+
+        // Met en surbrillance le point sélectionné
+        selectAll(".bullet").forEach((bullet, index) => {
+            if (index === activeIndex) {
+                bullet.classList.add("active");
+            } else {
+                bullet.classList.remove("active");
+            }
+        });
+    };
+
+
+    options.forEach((option, index) => {
+        option.addEventListener("click", () => handleClick(index));
+    });
+
+
+    function generateBullets(visibleOptions) {
+        bulletsContainer.innerHTML = "";
+
+
+        // New bullet point for each pictures
+        visibleOptions.forEach((option) => {
+            const bullet = document.createElement("span");
+            bullet.classList.add("bullet");
+            bullet.dataset.index = options.indexOf(option);
+            bullet.addEventListener("click", () =>
+                handleClick(options.indexOf(option))
+            );
+            bulletsContainer.appendChild(bullet);
+        });
+    }
```

Comment:
Not a huge fan of declaring a named function in the scope of an anonymous one... I know it work but it's a bit messy